simpleclient_spring_hibernate/src/main/java/io/prometheus/client/spring/hibernate/SimpleClientHibernateApplicationListener.java

```diff
@@ -0,0 +1,48 @@
+package io.prometheus.client.spring.hibernate;
+
+import io.prometheus.client.hibernate.HibernateStatisticsCollector;
+import org.hibernate.Session;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+import javax.persistence.EntityManager;
+
+@Component
+public class SimpleClientHibernateApplicationListener implements ApplicationListener<ApplicationReadyEvent> {
+  @Autowired
+  private EntityManager entityManager;
+
+  @Value("${spring.application.name:bootstrap}")
+  private String appName = "bootstrap";
+
+  private HibernateStatisticsCollector collector = new HibernateStatisticsCollector();
+
+  public SimpleClientHibernateApplicationListener() {
+
+  }
+
+  public SimpleClientHibernateApplicationListener(
+      HibernateStatisticsCollector collector,
+      EntityManager entityManager,
+      String appName
+  ) {
+    this.collector = collector;
+    this.entityManager = entityManager;
+    this.appName = appName;
+  }
+
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent event) {
+    if (entityManager == null) {
+      return;
+    }
+
+    Session session = (Session) entityManager.getDelegate();
+
+    collector.add(session.getSessionFactory(), appName);
+    collector.register();
```

Comment:
full disclosure, the application i've integrated is using SpringBoot `1.5.x`.

i'm working on an integration test right now to replace my mock-heavy test. i'm not so sure my 1.5 integration will work with a 1.3 app. i'll keep poking at it. ideally i'll come up with a solution for this repo that works with spring boot >= 1.3, instead of just >= 1.5.